python/lsst/resources/s3.py

```diff
@@ -234,6 +236,56 @@ def _mexists(
 
         return super()._mexists(uris, num_workers=num_workers)
 
+    @classmethod
+    def _mremove(cls, uris: Iterable[ResourcePath]) -> dict[ResourcePath, MBulkResult]:
+        # Delete multiple objects in one API call.
+        # Must group by profile and bucket.
+        grouped_uris: dict[tuple[str | None, str], list[S3ResourcePath]] = defaultdict(list)
+        for uri in uris:
+            uri = cast(S3ResourcePath, uri)
+            grouped_uris[uri._profile, uri._bucket].append(uri)
+
+        results: dict[ResourcePath, MBulkResult] = {}
+        for related_uris in grouped_uris.values():
+            # The client and bucket are the same for each of the remaining
+            # URIs.
+            first_uri = related_uris[0]
+            # API requires no more than 1000 per call.
+            for chunk in chunk_iterable(related_uris, chunk_size=1_000):
+                key_to_uri: dict[str, ResourcePath] = {}
+                keys: list[dict[str, str]] = []
+                for uri in chunk:
+                    key = uri.relativeToPathRoot
+                    key_to_uri[key] = uri
+                    keys.append({"Key": key})
+                    # Default to assuming everything worked.
+                    results[uri] = MBulkResult(True, None)
+                keys = [{"Key": uri.relativeToPathRoot} for uri in chunk]
```

Comment:
This looks like it duplicates what is done above?

Reply:
Yes. Thanks. I added the loop above and forgot to remove this line.